src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/clientpolicy/executor/UDSClientPolicyPermissionsExecutor.java

```diff
@@ -0,0 +1,132 @@
+                        logger.debug("Updating existing Client with Client ID: {}", clientCRUDContext.getTargetClient().getClientId());
+                        if (!isOwnedByUDSOperator(clientCRUDContext.getTargetClient()) && !canAccessWithInBackwardsCompatibilityMode(clientCRUDContext.getTargetClient())) {
+                            throw new ClientPolicyException(Errors.UNAUTHORIZED_CLIENT, "The Client doesn't have the " + ATTRIBUTE_UDS_OPERATOR + "=" + ATTRIBUTE_UDS_OPERATOR_VALUE + " attribute. Rejecting request.");
+                        }
+                        enforceClientSettings(clientCRUDContext.getProposedClientRepresentation());
+                        break;
+
+                    case REGISTER:
+                        logger.debug("Creating new Client with Client ID: {}", clientCRUDContext.getProposedClientRepresentation().getClientId());
+                        enforceClientSettings(clientCRUDContext.getProposedClientRepresentation());
+                        break;
+
+                    case VIEW:
+                    case UNREGISTER:
+                        logger.debug("Viewing or deleting Client with Client ID: {}", clientCRUDContext.getTargetClient().getClientId());
+                        if (!isOwnedByUDSOperator(clientCRUDContext.getTargetClient()) && !canAccessWithInBackwardsCompatibilityMode(clientCRUDContext.getTargetClient())) {
+                            throw new ClientPolicyException(Errors.UNAUTHORIZED_CLIENT, "The Client doesn't have the " + ATTRIBUTE_UDS_OPERATOR + "=" + ATTRIBUTE_UDS_OPERATOR_VALUE + " attribute. Rejecting request.");
+                        }
+                        break;
+                }
+            }
+        }
+    }
+
+    String getAuthenticatedClientId(ClientCRUDContext clientCRUDContext) {
+        if (clientCRUDContext.getAuthenticatedClient() == null) {
+            return null;
+        }
+        return clientCRUDContext.getAuthenticatedClient().getClientId();
+    }
+
+    boolean isOwnedByUDSOperator(ClientModel client) {
+        return ATTRIBUTE_UDS_OPERATOR_VALUE.equals(client.getAttribute(ATTRIBUTE_UDS_OPERATOR));
+    }
+
+    boolean canAccessWithInBackwardsCompatibilityMode(ClientModel client) {
+        switch (client.getClientId()) {
+            case "account":
+            case "account-console":
+            case "admin-cli":
+            case "broker":
+            case "realm-management":
+            case "security-admin-console":
+            case "uds-operator":
+            logger.warn("Denying access to the Client with Client ID: {}. This Client is not allowed to be accessed by the UDS Operator.", client.getClientId());
+            return false;
+        }
+
+        if (!client.getClientId().startsWith("uds")) {
+            logger.debug("Allowing access to the Client with Client ID: {} in backwards compatibility mode. Please note, that this type of access will be denied in the future.", client.getClientId());
+        }
```

Comment:
Okay leaving this as-is for now if you still see a use-case for it, but with the changes on the core side to add this attribute in dynamic mode I think we can avoid this backwards compatibility check?

Reply:
After thinking about this for a while, I think you're right. During the upgrade, Mission Heroes will first upgrade the UDS Operator, which will go through all the Packages and update the required Client Attribute. Then, Mission Heroes will start upgrading the Realm. 

I'll remove the backwards compatibility check but will also mention in the upgrade notes that if Mission Heroes hit any problems, it is advisable to disable the Policy and just it a bit more time to update everything.